tests/test_component_media.py

```diff
@@ -415,41 +415,19 @@ def test_glob_pattern_relative_to_root_dir(self):
         assertInHTML('<script src="glob/glob_2.js"></script>', rendered)
 
     def test_glob_pattern_does_not_break_urls(self):
-        class MyComponent(Component):
```

Comment:
This is why the edge case slipped the test - When a component is defined within a test file, the test file is not within the directories in `COMPONENTS.dirs`, and so the file paths are not even attempted to be resolved.

So after I moved the component class to be defined in a separate file, which is within `COMPONENTS.dirs`, then that triggered to try to resolve the file paths relative to the component file, which revealed the bug.